version: '3.7'

networks:
  private:
    driver: bridge

services:
  # Database
  postgres:
    image: postgres:11.7-alpine
    volumes:
      - /var/opt/api_local/pg_data:/var/lib/postgresql/data/pgdata
    environment:
      - PGDATA=/var/lib/postgresql/data/pgdata
      - POSTGRES_HOST_AUTH_METHOD=trust
    ports:
      - "5432:5432"
    networks:
      - private
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres"]
      interval: 30s
      timeout: 30s
      retries: 3
    restart: on-failure

  # Grafana
  grafana:
    build:
      context: ../grafana/
      dockerfile: Dockerfile
    env_file:
      - ../grafana/ENV/grafana.env
    volumes:
      - ./../grafana/ENV/grafana.ini:/etc/grafana/grafana.ini
    depends_on:
      - postgres
    ports:
      - "3001:3000"
    networks:
      - private
    restart: on-failure

  # Backend
  backend_demo_api:
    build:
      context: ../../apis/demo_api/
      dockerfile: Dockerfile
    env_file:
      - ../../apis/demo_api/ENV/api.env
    depends_on:
      - postgres
    ports:
      - "5001:5000"
    networks:
      - private
    restart: on-failure

  # Nodered
  node-red:
    image: vimagick/node-red
    environment:
      - TZ=Europe/Madrid
    volumes:
      - ./../../nodered/data:/data
    depends_on:
      - postgres
    ports:
      - "1880:1880"
    networks:
      - private
    restart: on-failure
